gax/src/main/java/com/google/api/gax/batching/DynamicFlowControlSettings.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.batching;
+
+import com.google.api.core.InternalApi;
+import com.google.api.gax.batching.FlowController.LimitExceededBehavior;
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Preconditions;
+import javax.annotation.Nullable;
+
+/** Settings for dynamic flow control */
+@AutoValue
+@InternalApi("For google-cloud-java client use only")
```

Comment:
That's not what this annotation is for: " Annotates a program element (class, method, package etc) which is internal to its containing library, not part of the public API, and should not be used by users of the library."

Adding it here risks tools flagging its use in google-cloud-java clients as incorrect, if not now, then in the future.

Reply:
For better or for worse, this is a pattern that is already established in gax-java & google-cloud-java clients. The reason for establishing this pattern is that gax serves 2 audiences: auto generated clients and end users. Currently gax doesnt have a formal way to target the different groups, so annotations are used.

Since this pattern is already actively used there are 3 paths forward:
1. invest time into splitting gax into gax-api for end users and gax-impl for google-cloud-java
2. Add a new annotation for InternalImplApi and migrate `@InternalApi("For google-cloud-java client use only")` to the new annotation
3. Have google-cloud-java tooling extract the message from InternalApi

None of that work is in scope for this PR. Feel free to ping me if you want to discuss further

Reply:
The most obvious path forward is to remove the annotation and accept that this is public API that must be supported for clients. 

This pattern might already be used, but this pattern is incorrect and does not do what it claims to do. Let's not introduce more instances of it.